packages/SwingSet/src/kernel/vatManager/factory.js

```diff
@@ -70,7 +70,7 @@ export function makeVatManagerFactory({
       !bundle || typeof bundle === 'object',
       `bundle must be object, not a plain string`,
     );
-    assert(!(setup && !enableSetup), `setup() provided, but not enabled`); // todo maybe useless
+    assert(!setup || enableSetup, X`setup() provided, but not enabled`); // todo maybe useless
```

Comment:
De Morgan and I are polite to one another at parties, but I value clarity over reduction, and in this case I think I prefer the original: it matches the english text directly. This is probably a case of Chip's positive-vs-negative issue, where this is sensing a badness condition (whose logic reads fairly naturally), rather than asserting a goodness condition (whose logic is a bit more contorted).

Reply:
hm, I almost want something that's the opposite of `assert`, like:

```js
deny(setup && !enableSetup, X`setup() provided but not enabled`);
```

to mean "here's something that should never happen, throw if you see it happening".

Reply:
Reverted this to the pre-de-morgan form